app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
+function findMatchingClosingBracketIndex(source, fromIndex) {
+  let result = -1
+  let currentDepth = 1
+  for (let i = fromIndex; i < source.length; i++) {
+    const char = source[i]
+    if (char === "{") {
+      currentDepth++
+    } else if (char === "}") {
+      currentDepth--
+      if (currentDepth <= 0) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
+  const conditionalStartRegex = /@if *\( *((?:.|\n)+?) *(==|!=) *((?:.|\n)+?)\) *[ \n]*\{/g
+  const conditionalElseStartRegex = / *@else *\{/
+
+  let match
+  while ((match = conditionalStartRegex.exec(source)) != null) {
+    const [matchedConditionalStartText, left, operation, right] = match
+    const afterMatchedTextIndex = match.index + matchedConditionalStartText.length
+
+    const matchingClosingBracketIndex = findMatchingClosingBracketIndex(source, afterMatchedTextIndex)
+    if (matchingClosingBracketIndex < 0) {
+      // TODO: warn user?
+      continue
+    }
+
+    let conditionalEndingIndex = matchingClosingBracketIndex
+
+    const trueBlockContent = source.substring(afterMatchedTextIndex, matchingClosingBracketIndex)
+    let falseBlockContent = ""
+
+    conditionalElseStartRegex.lastIndex = matchingClosingBracketIndex
+    const elseMatch = conditionalElseStartRegex.exec(source)
+    if (elseMatch != null) {
+      const afterElseMatchedTextIndex = elseMatch.index + elseMatch[0].length
+      const matchingClosingBracketForElseIndex = findMatchingClosingBracketIndex(source, afterElseMatchedTextIndex)
+      if (matchingClosingBracketForElseIndex > 0) {
+        falseBlockContent = source.substring(afterElseMatchedTextIndex, matchingClosingBracketForElseIndex)
+        conditionalEndingIndex = matchingClosingBracketForElseIndex
+      } // TODO: else warn user?
+    }
+
+    let passed = null
+    const sanitizedLeft = left.trimEnd()
+    const sanitizedRight = right.trimStart()
+    switch (operation) {
```

Comment:
We don't use switch cases anywhere. Not really anything against it, but perhaps it would be better to not do it here either. Mostly because there's 2 cases here which might be much nicer as an if statement.
```
if (operation == "==") passed = sanitizedLeft == sanitizedRight
else if (opeartion == "!=") passed = sanitizedLeft != sanitizedRight
```

Reply:
I used `switch` so in the future we could expand (see comment below the last `case`)

Reply:
Perhaps a different solution could be something like
```js
const operations = {
  "==": (l, r) => l == r,
  "!=": (l, r) => l != r
 }
 
passed = operations[operation]?.(sanitizedLeft, sanitizedRight)
 ```
 
 Which could turn in to
 ```js
const operations = {
  "is": (l, r) => l == r,
  "is not": (l, r) => l != r,
  "contains": (l, r) => l.contains(r)
 }
 ```